src/utils/utils.js

```diff
@@ -162,16 +170,16 @@ const PATH_END_REASONS = {
  * @param {boolean} deepOnly
  * @param {boolean} inline
  */
-export function createDOMPathGenerator(direction, deepOnly, inline, inScope = false) {
+export function createDOMPathGenerator(direction, deepOnly, inline, inScope = false, stopFunction) {
```

Comment:
Come to think of it, it feels weird to have a `inScope` argument along with a `stopFunction` argument. Shouldn't `inScope` be in charge of making sure we don't cross over boundaries that could be defined by a `stopFunction` ? In other words, isn't your scope actually defined by whatever is most restrictive, `stopFunction` included? These are genuine questions.

Reply:
Good question.
I am not sure why inScope was created and which case it protect but it is only used in `getNormalizedCursorPosition` through the call of now `leftDeepOnlyInlineInScopeNoEditable` and `rightDeepOnlyInlineInScopePath`. I don't see any other places where inScode is used.
What do you want me to do ? I can do one of:
- let it be like now
- check the contenteditable when inScope is defined
- check `inScope && currentNode === node` in stopFunction but:
  - the check is currently not done in `firstNode` nor `nextDeepest` so it would logically be different
  - the function `stopFunction` would require two arguments rather than one (ie. `currentNode` and `node` rather than only `currentNode`).
  - 
I have no opinion on this one, I let you chose.

Reply:
I'd say this gets slightly out of _scope_ because we're not sure of the effect it could have. I would however leave a note here (docstring ? :eyes:) to give an indication of what makes `inScope` different from `stopFunction`, and maybe create a non-prioritary task to investigate this someday.

Reply:
The thing is that I don't know what inScope is exactly supposed to cover.

Reply:
Then as it's only of secondary importance I would leave it at that but create the task so we keep it in mind for later.

Reply:
Task 2584236 created.